File: dotnet/samples/02-agents/providers/AnthropicProvider.cs
// Copyright (c) Microsoft. All rights reserved.

// Provider: Anthropic
// Create an agent using Anthropic (Claude) as the backend.
// Supports direct Anthropic API or Azure AI Foundry-hosted Anthropic models.
//
// For docs: https://learn.microsoft.com/agent-framework/agents/providers

using Anthropic;
using Anthropic.Foundry;
using Azure.Identity;
using Microsoft.Agents.AI;

string deploymentName = Environment.GetEnvironmentVariable("ANTHROPIC_DEPLOYMENT_NAME") ?? "claude-haiku-4-5";

// The resource is the subdomain in the Azure AI endpoint URI
string? resource = Environment.GetEnvironmentVariable("ANTHROPIC_RESOURCE");
string? apiKey = Environment.GetEnvironmentVariable("ANTHROPIC_API_KEY");

// <anthropic>
using AnthropicClient client = (resource is null)
    ? new AnthropicClient() { ApiKey = apiKey ?? throw new InvalidOperationException("ANTHROPIC_API_KEY is required when no ANTHROPIC_RESOURCE is provided") }
    : (apiKey is not null)
        ? new AnthropicFoundryClient(new AnthropicFoundryApiKeyCredentials(apiKey, resource))
        : new AnthropicFoundryClient(new AnthropicFoundryIdentityTokenCredentials(new AzureCliCredential(), resource, ["https://ai.azure.com/.default"]));

AIAgent agent = client.AsAIAgent(model: deploymentName, instructions: "You are good at telling jokes.", name: "JokerAgent");
// </anthropic>

Console.WriteLine(await agent.RunAsync("Tell me a joke about a pirate."));

File: python/samples/02-agents/providers/anthropic_provider.py
# Copyright (c) Microsoft. All rights reserved.

import asyncio

from agent_framework.anthropic import AnthropicClient

"""
Anthropic Provider

Demonstrates setting up AnthropicClient and running a simple query.

Environment variables:
- ANTHROPIC_API_KEY: Your Anthropic API key

For more Anthropic examples:
- With tools: getting_started/agents/anthropic/anthropic_claude_with_tools.py
- With MCP: getting_started/agents/anthropic/anthropic_claude_with_mcp.py
- Docs: https://learn.microsoft.com/agent-framework/providers/anthropic
"""


async def main() -> None:
    print("=== Anthropic Provider ===\n")

    # <create_agent>
    agent = AnthropicClient().as_agent(
        name="ClaudeAgent",
        instructions="You are a helpful assistant.",
    )
    # </create_agent>

    # <run_query>
    query = "What is the capital of France?"
    print(f"User: {query}")
    result = await agent.run(query)
    print(f"Agent: {result}")
    # </run_query>


if __name__ == "__main__":
    asyncio.run(main())
